lib/aws/broker/subscriber.rb

```diff
@@ -0,0 +1,82 @@
+require 'aws/broker/base'
+
+module Aws
+  class Broker
+    class Subscriber < Base
+
+      def initialize(topic, queue=nil)
+        @topic = topic
+        @queue = queue || queue_name
+      end
+
+      def subscribe
+        return unless enabled?
+        create_queue
+        create_topic
+        sns.subscribe(
+          topic_arn: @topic_arn,
+          protocol:  'sqs',
+          endpoint:  @queue_arn
+        )
+        set_queue_policy
+      end
+
+    private
+
+      def queue_name
+        if config.queue_prefix
+          "#{config.queue_prefix}-#{@topic}"
+        else
+          @topic
+        end
+      end
+
+      def create_queue
+        @queue_url = sqs.create_queue(
+          queue_name: @queue
+        ).queue_url
+        @queue_arn = sqs.get_queue_attributes(
+          queue_url:       @queue_url,
+          attribute_names: ['QueueArn']
+        ).attributes['QueueArn']
+      end
+
+      def set_queue_policy
+        sqs.set_queue_attributes(
+          queue_url:  @queue_url,
+          attributes: { 'Policy': policy }
+        )
+      end
+
+      def policy
+        <<-POLICY
+{
```

Comment:
Can this block be indented? I am sure there is something with spacing

Reply:
let me verify

Reply:
works with an indented policy! updated